doc/nrf-bm/libraries/bluetooth/ble_scan.rst

```diff
@@ -0,0 +1,261 @@
+           .scan_param = BLE_SCAN_INIT_SCAN_PARAM_DEFAULT,
+           .conn_param = BLE_SCAN_INIT_CONN_PARAM_DEFAULT,
+           .evt_handler = scan_event_handler_func,
+   };
+
+   nrf_err = ble_scan_init(&ble_scan, &scan_cfg);
+   if (nrf_err) {
+           LOG_ERR("Failed to initialie scan module, nrf_error %#x", nrf_err);
+   }
+
+
+Advanced Initialization
+=======================
+
+The advanced initialization provides a larger configuration set for the scan library.
+It is required when using scan filters or the whitelist.
+
+.. code:: c
+
+   BLE_SCAN_DEF(ble_scan);
+
+   void scan_event_handler_func(struct ble_scan_evt const *scan_evt);
+
+   uint32_t nrf_err;
+   struct ble_scan_config scan_cfg = {
+           .scan_param = {
+                   .active = 0x01,
+                   .interval = NRF_BLE_SCAN_SCAN_INTERVAL,
+                   .window = NRF_BLE_SCAN_SCAN_WINDOW,
+                   .filter_policy = BLE_GAP_SCAN_FP_WHITELIST,
+                   .timeout = SCAN_DURATION_WHITELIST,
+                   .scan_phys = BLE_GAP_PHY_1MBPS,
+                   .extended = true,
+           },
+           .conn_param = BLE_SCAN_INIT_CONN_PARAM_DEFAULT,
+           .connect_if_match = true,
+           .conn_cfg_tag = APP_BLE_CONN_CFG_TAG,
+           .evt_handler = scan_event_handler_func,
+   };
+
+   nrf_err = ble_scan_init(&ble_scan, &scan_cfg);
+   if (nrf_err) {
+           LOG_ERR("Failed to initialie scan module, nrf_error %#x", nrf_err);
+   }
+
+.. note::
+
+   When setting connection-specific configurations with the :c:func:`sd_ble_cfg_set` function, you must create a tag for each configuration.
+   If your application uses the Scan library with automatic connection, this tag must be provided when calling the :c:func:`sd_ble_gap_scan_start` or the :c:func:`sd_ble_gap_connect` function.
+
+Usage
+*****
+
+The application can start scanning by calling the :c:func:`ble_scan_start` function.
+
+The scan parameters can be updated by calling the :c:func:`ble_scan_params_set` function.
+
+To stop scanning, call the :c:func:`ble_scan_stop` function.
+
+The Scan library resumes scanning after receiving advertising reports.
+Scanning stops if the module establishes a connection automatically, or if the application calls the :c:func:`ble_scan_stop` or the :c:func:`sd_ble_gap_connect` function.
+
+.. note::
+
+   When you use the :c:func:`ble_scan_params_set` function during the ongoing scan, scanning is stopped.
+   To resume scanning, use the :c:func:`ble_scan_start` function.
+
+Whitelist
+=========
+
+The whitelist stores information about all the device connections and bonding.
+If you enable the whitelist, the application receives advertising packets only from the devices that are on the whitelist.
+An advertising package from a whitelisted device generates the :c:macro:`NRF_BLE_SCAN_EVT_WHITELIST_ADV_REPORT` event.
+
+.. note::
+
+   When using the whitelist, filters are inactive.
+
+.. warning::
```

Comment:
```suggestion
.. caution::
```
We do not use warnings. They are used only when an operating procedure or practice, which, if not correctly followed, could result in personal injury or loss of life.